<launch>

  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <arg name="moco_chain" default="MSA6_1001" />

  <!-- Load URDF -->
  <param name="robot_description" textfile="$(find motive_description)/urdf/msa6_v03.urdf" />

  <!-- Load hardware interface -->
  <node name="moco_hardware_interface" pkg="moco_control" type="moco_hw_main"
           output="screen" launch-prefix="$(arg launch_prefix)"/>

  <!-- Load controller settings -->
  <rosparam file="$(find moco_control)/config/moco_controllers.yaml" command="load"/>

  <!-- Convert joint states to /tf tranforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

  <!-- Load MOCO-related parameters -->
  <rosparam file="$(find moco_control)/config/moco_controllers.yaml" command="load"/>
  <param name="hardware_interface/chain" type="string" value="$(arg moco_chain)" />

  <!-- Spawn and start these controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="joint_state_controller arm_trajectory_controller" />

  <!-- Load other controllers -->
  <node name="controller_loader" pkg="controller_manager" type="controller_manager" respawn="false"
    output="screen" args="load streaming_position_controller" />
</launch>
